resources/prosody-plugins/mod_visitors.lua

```diff
@@ -240,6 +240,36 @@ process_host_module(main_muc_component_config, function(host_module, host)
         end
     end);
 
+    -- when a main participant is kicked inform the visitor nodes
+    host_module:hook('muc-broadcast-presence', function(event)
+        local x, actor = event.x, event.actor;
+        if not actor or not presence_check_status(x, '307') then
```

Comment:
I wonder about performance here. 
Maybe we can just create a presence unavailable here: https://github.com/jitsi/jitsi-meet/blob/8a2e4bc6281ee2d7a0602d359a03c8dc9c2c34e0/resources/prosody-plugins/mod_visitors.lua#L239 and send it, instead of the log...
Although I see one more place where we use it and probably we can do some optimization to skip this check ...

Reply:
https://github.com/bjc/prosody/blob/cbcb57fa24650c6ac001c4c1c9ab39a61b6266eb/plugins/muc/muc.lib.lua#L231

Reply:
WDYT?

Reply:
> I wonder about performance here. Maybe we can just create a presence unavailable here:
> 
> https://github.com/jitsi/jitsi-meet/blob/8a2e4bc6281ee2d7a0602d359a03c8dc9c2c34e0/resources/prosody-plugins/mod_visitors.lua#L239
> 
> and send it, instead of the log...
> Although I see one more place where we use it and probably we can do some optimization to skip this check ...

This makes more sense.I will test and push it.